database-plugins/src/main/java/io/cdap/plugin/db/batch/source/DBSource.java

```diff
@@ -357,4 +358,16 @@ private Schema getSchema(FailureCollector collector) {
       throw collector.getOrThrowException();
     }
   }
+
+  private void emitLineage(BatchSourceContext context) {
+    Schema schema = sourceConfig.getSchema(context.getFailureCollector());
```

Comment:
config.getSchema() can return null, if it is null, you should get it from `context.getOutputSchema`